### PR TITLE
Remove dependency on cats-tagless

### PR DESCRIPTION
**Pull request by jchapuis**
Sacrifice a bit of abstraction for simplicity and the ability to cross-build to Scala 3. This was going to break compatibility anyway so I also took advantage of clarifying some names.

Main changes are:
 - protocol client has to invoke the command sender directly (renamed from router) passed in implicit scope, rather than this sending aspect added via an interpreter. But this is made easy thanks to a `sendCommand` helper function.

 - effector logic is no longer run with a monad transformer, `EffectorT` disappears. This is a big gain in simplicity. I realized that since we need to support `Async` anyway, we can just implement effector functionality with cats-effect primitives

### CODE REVIEW

**Comment by jchapuis** [comment]
I realized I can still clarify naming of the core concepts in 'deployRepository', I'm now thinking of 

'''scala
repository: Repository(sharding)
behavior: Behavior(entity)
effect: Effect(effector)
'''

**Comment by jchapuis** [comment]
> I realized I can still clarify naming of the core concepts in 'deployRepository', I'm now thinking of
> 
> repository: Repository(sharding) behavior: Behavior(entity) effect: Effect(effector)

I have added a commit with further renamings and changes that hopefully improve ergonomics, at least I'm quite happy with them 😃

File: akka-runtime/src/main/scala/endless/runtime/akka/deploy/AkkaDeployer.scala [hunk 1]
```diff
@@ -2,41 +2,43 @@ package endless.runtime.akka.deploy
...
 
-  def deployRepository[F[_]: Async, ID: EntityIDCodec, S, E, Alg[_[_]]: FunctorK, RepositoryAlg[_[
+  override def deployRepository[F[_]: Async, ID: EntityIDCodec, S, E, Alg[_[_]], RepositoryAlg[_[
       _
   ]]](
       repository: RepositoryInterpreter[F, ID, Alg, RepositoryAlg],
-      entity: EntityInterpreter[F, S, E, Alg],
-      effector: EffectorInterpreter[F, S, Alg, RepositoryAlg]
+      behavior: BehaviorInterpreter[F, S, E, Alg],
+      effector: SideEffectInterpreter[F, S, Alg, RepositoryAlg]
```
Comment (amiguez) [inline akka-runtime/src/main/scala/endless/runtime/akka/deploy/AkkaDeployer.scala:29]: nit: shall we also name this `sideEffect` as for the Durable Entity?

**Review by amiguez — APPROVED** [review]
LGTM :100:

File: example/src/main/scala/endless/example/app/pekko/PekkoApp.scala [hunk 2]
```diff
@@ -111,16 +111,16 @@ object PekkoApp extends Bookings with Vehicles with Availabilities {
...
                   BookingAlg,
-                  BookingRepositoryAlg
+                  BookingsAlg
                 ](
-                  RepositoryInterpreter.pure(BookingRepository(_)),
-                  EntityInterpreter.pure(BookingEntity(_)),
-                  EffectorInterpreter.pure { case (effector, _, _) => BookingEffector(effector) }
+                  RepositoryInterpreter.lift(ShardedBookings(_)),
+                  BehaviorInterpreter.lift(BookingEntity(_)),
+                  SideEffectInterpreter.lift { case (_, _) => BookingSideEffect() }
```
Comment (amiguez) [inline example/src/main/scala/endless/example/app/pekko/PekkoApp.scala:118]: I indeed like the renaming :+1:

File: akka-runtime/src/main/scala/endless/runtime/akka/deploy/AkkaDeployer.scala (hunk 1 above)
Comment (jchapuis) [reply akka-runtime/src/main/scala/endless/runtime/akka/deploy/AkkaDeployer.scala:29]: yes, thanks 👍

**Pull request merged**